base/special/exp.jl

```diff
@@ -275,17 +275,18 @@ end
     r = muladd(N_float, LogBU(base, T), x)
     r = muladd(N_float, LogBL(base, T), r)
     small_part = expb_kernel(base, r)
-    if !(abs(x) <= SUBNORM_EXP(base, T))
-        x > MAX_EXP(base, T) && return Inf32
-        x < MIN_EXP(base, T) && return 0.0f0
-        if N <= Int32(-24)
-            twopk = reinterpret(T, (N+Int32(151)) << Int32(23))
-            return (twopk*small_part)*(2f0^(-24))
-        end
-        N == 128 && return small_part * T(2.0) * T(2.0)^127
+    power = (N+Int32(127))
+    x > MAX_EXP(base, T) && return Inf32
+    x < MIN_EXP(base, T) && return 0.0f0
+    if x <= -SUBNORM_EXP(base, T)
```

Comment:
So in the non-vectorized case you have 3 comparisons instead of 1 in the typical case?  Does that slow down non-vectorized usage?

Reply:
Accurately benchmarking this is rather difficult because it depends a lot on the branch predictor. That said, for a simple test, of `julia> @btime exp10(x) setup=(x=randn()*100)`, I get `4.870 ns` with the branch compared to `4.840 ns (0 allocations: 0 bytes)` on master.